plugins/database/mongodb/connection_producer.go

```diff
@@ -106,120 +135,49 @@ func (c *mongoDBConnectionProducer) Init(ctx context.Context, conf map[string]in
 
 // Connection creates or returns an existing a database connection. If the session fails
 // on a ping check, the session will be closed and then re-created.
-func (c *mongoDBConnectionProducer) Connection(_ context.Context) (interface{}, error) {
+func (c *mongoDBConnectionProducer) Connection(ctx context.Context) (interface{}, error) {
 	if !c.Initialized {
```

Comment:
Should we lock the Mutex in this function as there are many reads and writes against `c.*`?

Reply:
`Connection` does not care about the lock but actually assumes that it is handled by the caller. This is also the case for other `connection_producer` from other DB plugins. I added a comment to make this more clear.

Reply:
👍